src/command_server.py

```diff
@@ -6,126 +6,124 @@
-                        'game': self._controller.game_field,
-                        'units': self._controller.get_units(),
-                        'frequency': STEP_INTERVAL}}
-                    yield from websocket.send(json.dumps(start_conditions))
-                    break
-        while websocket.open:
-            data = yield from websocket.recv()
-            if data:
-                data = json.loads(data)
-                for key in data:
-                    action = getattr(my_hero, key, '')
-                    if action:
-                        action(data[key])
-            yield from asyncio.sleep(STEP_INTERVAL)
-        if self._controller.units.get(my_hero.id):
-            self._controller.remove_unit(my_hero.id)
-        yield from websocket.close()
 
-    @asyncio.coroutine
-    def notify_clients(self, data):
-        if hasattr(self._server, 'websockets'):
-            for socket in self._server.websockets:
-                yield from socket.send(json.dumps(data))
-
-
-class HttpCommandServer(BaseCommandServer):
-    _instance = None
-
-    def _init_server(self, host, port):
-        self._app = web.Application()
         self._load_routes()
         self._load_static()
+        self._controller = get_game(600, 800, 5)
         self._server = self._loop.create_server(self._app.make_handler(),
-            host, port)
+                                                host, port)
+
+        if templates:
+            aiohttp_jinja2.setup(self._app,
+                                 loader=jinja2.FileSystemLoader(templates))
 
     def __new__(cls, *args, **kwargs):
         if not cls._instance:
             cls._instance = super(HttpCommandServer, cls).__new__(cls)
         return cls._instance
 
+    def start(self):
+        self._server = self._loop.run_until_complete(self._server)
+        logging.info('Server has started.')
+
+    def stop(self):
+        self._server.close()
+        logging.info('Server has stopped.')
+
     def _load_routes(self):
         logging.debug('Loading  Application Routes:\n%s' % '\n'.join(str(r) for r in routes.ROUTES))
+        self._app.router.add_route('GET', '/ws_stream', self.ws_stream)
         for route in routes.ROUTES:
             self._app.router.add_route(*route)
 
     def _load_static(self):
         self._app.router.add_static('/static', static_path)
 
+    @asyncio.coroutine
+    def ws_stream(self, request, *args, **kwargs):
+        ws = web.WebSocketResponse()
+        ws.start(request)
+
+        while ws.started:
+            msg = yield from ws.receive()
+            if msg.tp == MsgType.text:
+                if msg.data == 'close':
+                    yield from ws.close()
+                    self._controller.del_web_socket(ws)
+                    if self._controller.units.get(my_hero.id):
+                        self._controller.remove_unit(my_hero.id)
+                else:
+                    data = json.loads(msg.data)
```

Comment:
All that logic must be part of GameController. HTTP server must know nothing about heroes, invaders, bullets and any other units with actions